### DISCUSSION TITLE
How to specify color for AppCSXCAD in Python API CSX.AddMaterial()?

### DISCUSSION THREAD

**Discussion by VolkerMuehlhaus** [thliebig/openEMS-Project#257] [Q&A Usage] [answered]
Dear all,

I want to control the color and opacity of materials shown in AppCSXCAD, and specify that in my Python code when creating the model.
Looking at the source code, I think that this is possible and CSX.AddMaterial() is the place to specify color and opacity for the material, but could not figure out the syntax.

Could anyone help me out?

Best regards
Volker

**Comment by thliebig** [comment]
Hello Volker,

I think you need to use the ["SetColor"](https://docs.openems.de/python/CSXCAD/CSProperties/CSProperties.html#CSXCAD.CSProperties.CSProperties.SetColor) method on the properties.
But at the moment this needs to be a hex value.

best regards
Thorsten

**Reply by VolkerMuehlhaus** [reply]
Hi Thorsten, 
yes this what I found, but I didn't understand how to actually use that in my code. Could you draft a quick example?

Here is what I have:
`CSX_material = CSX.AddMaterial(material.name, kappa=material.sigma, epsilon=material.eps)`
Now where/how do I use that Setcolor method, to set color 0xff0000 ?

Many thanks!
Volker

**Reply by thliebig** [reply, marked as answer]
"CSX_material" is here your properties instance on which you can call all the method available by the interface.
This means in your case:
`CSX_material.SetColor('#ff0000')`

**Reply by VolkerMuehlhaus** [reply]
Many thanks, Thorsten! I had tried some things, but didn't find the actual syntax. Works fine now!
